CONTRIBUTION.md

```diff
@@ -28,58 +28,126 @@ We know that as of 1.12 this is not the case for great parts of the mod, but we
 Unless your PR is directly related to code cleaning then your PR should not contain any formatting changes to unchanged code blocks as this simply makes reviewing them unnecessarily complex.
 
 ### Formatting
-* Use 4 space indent instead of Tabs
-* operators should be surrounded by spaces
-  * For example `if (statement == true)`
-* a `,` comma to separate types should be followed by exactly one space
-* type casts should be followed by exactly one space
-* `{` should not be prefixed with returns
-* Methods should have exactly one free line between them (or their doc blocks)
-* Avoid empty lines at the start of class, interface or enum definitions
-* Avoid unnecessary lines in empty Methods
-    * `public void emptyImpl() {}`
-    * Also try and avoid introducing code that may require empty methods
-* Method/Class annotations should gain their own lines in front of the definition
-* Parameter Annotations should be surrounded by exactly one space
-* Comments should always start with a single space `// Comment`
+* Use 4 space indent instead of Tabs.
+* Operators should be surrounded by spaces:
+  * For example `if (statement == true)`.
+* A `,` comma to separate types should be followed by exactly one space.
+* Type casts should be followed by exactly one space.
+* `{` should not be wrapped onto a new line.
+* Methods should have exactly one free line between them (or their doc blocks).
+* Avoid empty lines at the start of class, interface or enum definitions.
+* Avoid unnecessary lines in empty Methods:
+  * `public void emptyImpl() {}`.
+  * Try to avoid introducing code that may require empty methods.
+* Method/Class annotations should gain their own lines in front of the definition.
+* Parameter Annotations should be surrounded by exactly one space.
+* Comments should always start with a single space `// Comment`.
+* Import statments should be ordered in sections with exactly 1 empty line between them. Regular imports go in the front and static imports follow them.
+  * If you are using IntelliJ, this is the default order.
+  * If you are using Eclipse, you need to reorder them under `Preferences -> Java -> Code Style -> Organize Imports`.
+    1. `com.`, `.net`, and other imports, should be in alphebetical order
+    2. (empty line)
+    3. `java.`, `javax.`, etc.
+    4. (empty line)
+    5. All other static imports
```

Comment:
Can't eclipse do the alphabetical thing?

Reply:
All unmatched type imports are alphabetically sorted (ad have no empty lines between them). If I don't specify `com`, `net`, or `org`, then they will be alphabetically sorted along with everything other than `java` and `javax` at the bottom.

If they are specified, then there will be an empty line between each of them.

Reply:
Also, what should the cutoff import number be to automatically use `*` (by default it's 99)?

Reply:
I set it to 3 and IntelliJ default is 5...

Reply:
There should be some standard number to agree upon and include here.

Reply:
I have to admit that I usually revert the auto-import for inner classes whilst at the same time enforcing static import of utility Methods already with the first Method... (simulating extension Methods)

Reply:
How about 3 for the cutoff number?

Reply:
So `*` would be used for `> 3` or `>= 3`?

Reply:
> 3

Reply:
So, let's say one takes this class https://github.com/Direwolf20-MC/BuildingGadgets/blob/master/src/main/java/com/direwolf20/buildinggadgets/client/RemoteInventoryCache.java and adds 2 different types of imports to it, besides `com`, `net`, `org`, `java`, and `javax`. After doing so, can someone please copy/paste the result of using `organize imports` on it, given:
1. The use of `*` with `> 3`.
2. All other import settings are at their default values for InteliJ.

Reply:
```java
import com.direwolf20.buildinggadgets.common.network.PacketHandler;
import com.direwolf20.buildinggadgets.common.network.PacketSetRemoteInventoryCache;
import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
import com.direwolf20.buildinggadgets.common.tools.InventoryManipulation.IRemoteInventoryProvider;
import com.direwolf20.buildinggadgets.common.tools.UniqueItem;
import com.google.common.base.Stopwatch;
import com.google.common.collect.Multiset;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.math.BlockPos;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import org.apache.commons.lang3.tuple.ImmutablePair;
import org.apache.commons.lang3.tuple.Pair;

import javax.annotation.Nullable;
import java.util.concurrent.TimeUnit;
```

Reply:
And how about when you also add "2 different types of imports to it, besides `com`, `net`, `org`, `java`, and `javax`"?

Reply:
```java
import com.direwolf20.buildinggadgets.common.integration.mods.SimpleStorageNetworkAPI;
import com.direwolf20.buildinggadgets.common.network.PacketHandler;
import com.direwolf20.buildinggadgets.common.network.PacketSetRemoteInventoryCache;
import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
import com.direwolf20.buildinggadgets.common.tools.InventoryManipulation.IRemoteInventoryProvider;
import com.direwolf20.buildinggadgets.common.tools.UniqueItem;
import com.google.common.base.Stopwatch;
import com.google.common.collect.Multiset;
import mekanism.api.MekanismAPI; // Example
import mrriegel.storagenetwork.api.IStorageNetworkPlugin; // Example
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.math.BlockPos;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import org.apache.commons.lang3.tuple.ImmutablePair;
import org.apache.commons.lang3.tuple.Pair;

import javax.annotation.Nullable;
import java.util.concurrent.TimeUnit;
```

Reply:
Where is `SimpleStorageNetworkAPI` coming from?

Reply:
The top line, `import com.direwolf20.buildinggadgets.common.integration.mods.SimpleStorageNetworkAPI;`. It is the top most line because `i`ntegration is before `n`etwork.

Reply:
Of course you can: `Class c = SimpleStorageNetworkAPI.class;`

Reply:
In any event; I'm happy with this as the standard:
1. Unmatched imports (alphabetical)
2. `java` & `javax` imports
3. Static imports

Sadly, Eclipse has no way of preventing empty lines from being added between specified import types, so when I use `organize imports`, there is an empty line between the `java` & `javax` imports. But that's the only difference, so it's not a big deal.